### DISCUSSION TITLE
Mac OSX build

### DISCUSSION THREAD

**Discussion by Mack-Mack** [swfans/syndwarsfx#199] [Q&A] [answered]
I would like to run the new port on my MacBook Air M2 and I have a few questions.

Firstly, is a compiled Mac OSX executable / bundle released either here or externally?

Secondly I have read the build instructions and I am considering attempting to compile it myself however while I am reasonably computer literate (familiar with DOS command line etc) I have never complied an executable before. Does anyone know if the compiled code will actually run on an Apple Silicon Mac? Also beyond the build instructions if anyone can recommend any guides etc to assist me in compiling the code?

Thanks for your time and for this awesome port.

**Comment by mefistotelis** [comment, marked as answer]
I don't use any Apple devices. The code was not tested on a Mac for a long time - the original SWPort 0.3 (which is the first commit in this repository) was built and tested on a Mac back in 2011 by original authors. But all the further changes were made without testing on a Mac.

From what I understand, devices with the ARM chips designed by Apple have a simulator which allows to run x86 applications. If that simulator supports 32-bit apps, then it will allow to run SW Port, compiled for x86.

It is not currently possible to compile SW Port for native ARM architecture, because of extensive use of x86 assembly within the project sources. So, you will still need a compiler toolchain for x86 Mac.

The compiling doesn't sound like an easy task for a newbie:
* It wasn't attempted for over a decade
* It requires cross-compilation with 32-bit x86 toolchain
* Build instructions were not updated by over a decade

If you still want to give it a try, I can help with any compile errors you will encounter. For how to prepare an environment and properly start the build - you'd have to ask someone who actually uses the Apple ecosystem.

**Reply by mefistotelis** [reply]
Checking what people write, the last version which supported building 32-bit software was *macOS 10.14*.
GitHub Actions support only MacOS 12 and up, so there will be no automatic building.

**Comment by Mack-Mack** [comment]
It seams the biggest stumbling block is the apparently inability to run 32 bit code on newer versions of OSX and or Apple Silicon. Given the already difficult process of compiling the code I will look into other options otherwise continue emulating under DOSBox staging. Thank you for taking the time to answer my query.
